lib/learning-object/learning-object.ts

```diff
@@ -142,195 +257,291 @@ export class LearningObject {
    *       reference itself is immutable, and elements can only be
    *       added and removed by the below functions
    */
-  goals: LearningGoal[];
+  get goals(): LearningGoal[] {
+    return this._goals;
+  }
+  /**
+   * Adds a new learning goal to this object.
+   * Returns the index of the new goal
+   */
+  addGoal(text: string): number {
+    const goal = new LearningGoal(text);
+    return this._goals.push(goal) - 1;
+  }
+  /**
+   * Removes the object's i-th learning goal.
+   * @param {number} index the index to remove from this object's goals
+   *
+   * @returns {LearningGoal} the goal which was removed
+   */
+  removeGoal(index: number): LearningGoal {
+    return this._goals.splice(index, 1)[0];
+  }
 
+  private _outcomes: LearningOutcome[];
   /**
    * @property {LearningOutcome[]} outcomes
    *       outcomes this object should enable students to achieve
    *
    */
-  outcomes: LearningOutcome[];
+  get outcomes(): LearningOutcome[] {
+    return this._outcomes;
+  }
+  /**
+   * Adds a passed outcome or new, blank learning outcome to this object.
+   * @returns {number} index of the outcome
+   */
+  addOutcome(outcome?: LearningOutcome): number {
+    const addingOutcome = outcome || new LearningOutcome();
+    return this._outcomes.push(addingOutcome) - 1;
+  }
+  /**
+   * Removes the object's i-th learning outcome.
+   * @param {number} index the index to remove from this objects' outcomes
+   *
+   * @returns {LearningOutcome} the learning outcome which was removed
+   */
+  removeOutcome(index: number): LearningOutcome {
+    return this._outcomes.splice(index, 1)[0];
+  }
 
+  private _materials!: Material;
   /**
    * @property {Material} materials neutrino file/url storage
    *
-   * TODO: extend constituents into full-fledged entities
    */
-  materials: Material;
+  get materials(): Material {
+    return this._materials;
+  }
+  set materials(material: Material) {
+    this._materials = material;
+  }
 
+  private _metrics!: Metrics;
   /**
    * @property {Metrics} metrics neutrino file/url storage
    *
-   * TODO: extend constituents into full-fledged entities
    */
-  metrics: Metrics;
+  get metrics(): Metrics {
+    return this._metrics;
+  }
+  set metrics(metrics: Metrics) {
+    this._metrics = metrics;
+  }
 
+  private _published: boolean;
   /**
    * @property {boolean} published
    *       Whether or not the Learning Object is published
    */
-  published: boolean;
-
+  get published(): boolean {
+    return this._published;
+  }
   /**
-   * Sets LearningObject's published flag to true
+   * Sets LearningObject's stats to published and published flag to true
    *
    * @memberof LearningObject
    */
   publish(): void {
-    this.published = true;
+    this._status = 'published';
```

Comment:
I don't think there's any need for the learning object to be set manually to 'published' as that bypasses the review process. Maybe 'waiting' instead?
```suggestion
     this._status = 'waiting';
```

Reply:
My thought was that any call to publish mean that the object has gone through the review process and is being "published" therefore the status is also set to published.